prov/efa/src/rxr/rxr_pkt_type_req.c

```diff
@@ -1052,6 +1020,52 @@ void rxr_pkt_handle_long_rtw_recv(struct rxr_ep *ep,
 	rxr_pkt_entry_release_rx(ep, pkt_entry);
 }
 
+void rxr_pkt_handle_read_rtw_recv(struct rxr_ep *ep,
+				  struct rxr_pkt_entry *pkt_entry)
+{
+	struct rxr_rx_entry *rx_entry;
+	struct rxr_read_rtw_hdr *rtw_hdr;
+	struct fi_rma_iov *read_iov;
+	ssize_t err;
+
+	rx_entry = rxr_pkt_alloc_rtw_rx_entry(ep, pkt_entry);
+	if (!rx_entry) {
+		FI_WARN(&rxr_prov, FI_LOG_CQ,
+			"RX entries exhausted.\n");
+		efa_eq_write_error(&ep->util_ep, FI_ENOBUFS, -FI_ENOBUFS);
+		return;
+	}
+
+	rtw_hdr = (struct rxr_read_rtw_hdr *)pkt_entry->pkt;
+	rx_entry->iov_count = rtw_hdr->rma_iov_count;
+	err = rxr_rma_verified_copy_iov(ep, rtw_hdr->rma_iov, rtw_hdr->rma_iov_count,
+					FI_RECV, rx_entry->iov);
+	if (err) {
+		FI_WARN(&rxr_prov, FI_LOG_CQ, "RMA address verify failed!\n");
```

Comment:
There's a case here (and in a few other places) where we don't return. That'll cause a crash later when the iov is NULL. Can you open another PR that addresses this?

Reply:
Sure, will open another PR

Reply:
I updated this PR with the fixes, also added call to post_connack()